### PR TITLE
feat: Added reading service account in Github action

### LINKED ISSUE #1597: Get and set service account
Add option to Github action to pass service account as a parameter, setup it correctly to allow running Flank.
Allow passing the path to file and file content itself(the best using single option)

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1597

### Flank scripts part
Run command
- `flankScripts firebase save_service_account --account=<file content, link, file path>` -> to test storing service account

### Github action part
You could run Flank GitHub action on [testing repository](https://github.com/Flank/flank-github-action-testing/actions)

## Checklist

- [x] Documented
- [x] Unit tested
- [x] Test repository updated

### CODE REVIEW

File: flank-scripts/build.gradle.kts [hunk 1]
```diff
@@ -26,7 +26,7 @@ shadowJar.apply {
     }
 }
 // <breaking change>.<feature added>.<fix/minor change>
-version = "1.8.0"
+version = "1.8.1"
```
Comment (piotradamczyk5) [inline flank-scripts/build.gradle.kts:29]:
```suggestion
version = "1.9.0"
```

File: action.yml [hunk 2]
```diff
@@ -17,3 +20,7 @@ runs:
       run: |
         ./flankScripts github download_flank --version=${{ inputs.version }}
       shell: bash
+    - id: authentication
+      run: |
+        ./flankScripts <TODO fill with command> --service_account=${{ inputs.service_account }}
```
Comment (piotradamczyk5) [inline action.yml:25]:
```suggestion
        ./flankScripts firebase save_service_account
 --account=${{ inputs.service_account }}
```

File: flank-scripts/build.gradle.kts (hunk 1 above)
Comment (piotradamczyk5) [reply flank-scripts/build.gradle.kts:29]: version up

**Review by Sloox — APPROVED** [review]

File: docs/flank-scripts/command_overview.md [hunk 3]
```diff
@@ -31,6 +31,7 @@ ___
   - `check_for_sdk_updates` - Check for new SDK features and create update tasks for it
     - `generate_client` - Generate Java Client based on api schema
     - `update_api` - Update api schema
+    - `save_service_account` - Save given service account to flank credentials location
```
Comment (pawelpasterz) [inline docs/flank-scripts/command_overview.md:34]: just indention

File: flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import flank.common.defaultCredentialPath
+import flank.common.downloadFile
+import java.io.File
+import java.nio.file.Path
+
+fun saveServiceAccount(
+    serviceAccount: String,
+    serviceAccountPath: Path = defaultCredentialPath.toAbsolutePath()
+) = when {
+    serviceAccount.startsWith("http", true) -> downloadFile(serviceAccount, serviceAccountPath)
```
Comment (pawelpasterz) [inline flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt:12]: Is it possible to pass service acc via HTTP request? (just curious)

File: flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import flank.scripts.ops.firebase.saveServiceAccount
+
+object SaveServiceAccountCommand : CliktCommand(
+    name = "save_service_account",
+    help = "Save given service account to flank credentials location"
+) {
+    private val serviceAccount: String by option("--account").required()
```
Comment (pawelpasterz) [inline flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt:12]: Do we want to validate it somehow? (for example non-empty string)

Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt:12]: And maybe nice error handling? (not request, just thinking)

File: flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt (hunk 4 above)
Comment (piotradamczyk5) [reply flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt:12]: if you have hosted in somewhere, I do not see a problem 😄

File: flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt (hunk 5 above)
Comment (piotradamczyk5) [reply flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt:12]: If you did not pass the parameter it should fail on `flank-scripts` side. Other validation will be performed on Flank side

File: flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt (hunk 4 above)
Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt:12]: Never said it's a problem, I was curious about use case

Comment (piotradamczyk5) [reply flank-scripts/src/main/kotlin/flank/scripts/ops/firebase/SaveServiceAccount.kt:12]: I mean that there is no problem to host a service account somewhere and read it later on

File: flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt (hunk 5 above)
Comment (pawelpasterz) [reply flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt:12]: Got it 👍 
There are two cases I meant:
* `firebase save_service_account --account=` will create an empty JSON file (the question is if we should bother 🤷 )
* `firebase save_service_account --account=asdasdsdsfas.json` throws an error with nasty stacktarce (same as above)

Comment (piotradamczyk5) [reply flank-scripts/src/main/kotlin/flank/scripts/cli/firebase/SaveServiceAccountCommand.kt:12]: Personally, I think that we should not validate the service account provided by user

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**